### PR TITLE
Arc-Docs  Introduction Part

### LINKED ISSUE #26: Documention of Boilerplate
Documentation related to how can we use backend and other things in boilerplate

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Deepika516**
## Description
created the standalone components (Sidebar and header) making the bolierplate Responsive as well and  Adding Documentation of Arc Boilerplate 

Fixes #26 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/arc/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -6,24 +6,24 @@ import {AuthGuard, LoggedInGuard} from '@project-lib/core/auth';
...
+  import('projects/arc-lib/src/lib/components/auth/auth.module').then(
+  m => m.AuthModule,
+  ),
+  // canActivate: [LoggedInGuard],
   },
   {
     path: 'main',
     loadChildren: () => import('./main/main.module').then(m => m.MainModule),
-    canActivate: [AuthGuard],
+    // canActivate: [AuthGuard],
```
Comment (yeshamavani) [inline projects/arc/src/app/app-routing.module.ts:18]: why commented while commiting
please recheck before pushing the code 
this is a wrong practice

File: projects/arc/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -6,24 +6,24 @@ import {AuthGuard, LoggedInGuard} from '@project-lib/core/auth';
...
   },
   {
     path: 'gantt',
     loadChildren: () =>
       import('../../../arc-lib/src/lib/components/gantt/gantt.module').then(
         m => m.GanttModule,
       ),
-    canActivate: [AuthGuard],
+    // canActivate: [AuthGuard],
```
Comment (yeshamavani) [inline projects/arc/src/app/app-routing.module.ts:26]: same

File: projects/arc/src/app/main/main.component.ts [hunk 3]
```diff
@@ -63,15 +63,6 @@ export class MainComponent
...
-  navigate(link: string) {
-    this.router.navigate([link]);
-  }
-
-  toggle() {
-    this.sidebarService.toggle(true, 'right');
-    this.toggleFooter = !this.toggleFooter;
-  }
-
   ngOnInit(): void {
```
Comment (yeshamavani) [inline projects/arc/src/app/main/main.component.ts:66]: why removed ?

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/arc-docs/src/app/docs/docs.module.ts [hunk 4]
```diff
@@ -3,9 +3,61 @@ import {CommonModule} from '@angular/common';
...
+    // AuthService,
+    // UserSessionStoreService,
+    // {provide: APPLICATION_STORE, useExisting: LOCAL_STORAGE},
+    // {provide: APP_SESSION_STORE, useExisting: SESSION_STORAGE},
+    // InMemoryStorageService,
+    // ApiService,
+    // SignUpAdapter,
+    // LoggedInUserAdapterService,
+    // LoginAdapterService,
+    // AnyAdapter,
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/docs.module.ts:55]: done

File: projects/arc/src/app/app-routing.module.ts (hunk 1 above)
Comment (Deepika516) [reply projects/arc/src/app/app-routing.module.ts:18]: my mistake it pushed like that i ll take care

File: projects/arc/src/app/app-routing.module.ts (hunk 2 above)
Comment (Deepika516) [reply projects/arc/src/app/app-routing.module.ts:26]: done

File: projects/arc-docs/src/app/docs/getting-started/components/introduction/introduction.component.html [hunk 5]
```diff
@@ -1 +1,138 @@
...
+        <h2 class="nb-card-header-title">What is Arc-Angular-Boilerplate</h2>
+      </nb-card-header>
+      <nb-card-body>
+        <p>
+          Arc-Boilerplate is a feature-rich library specifically designed to
+          enhance your Angular development experience. It provides a curated
+          collection of Angular components, directives, services, and utilities,
+          all carefully crafted to accelerate your development workflow and
+          promote best practices.
+        </p>
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/getting-started/components/introduction/introduction.component.html:14]: Done

File: projects/arc-docs/src/app/docs/getting-started/components/introduction/introduction.component.html [hunk 6]
```diff
@@ -1 +1,138 @@
...
+      <nb-card-header fixed>
+        <h2 class="nb-card-header-title">What is Arc-Angular-Boilerplate</h2>
+      </nb-card-header>
+      <nb-card-body>
+        <p>
+          Arc-Boilerplate is a feature-rich library specifically designed to
+          enhance your Angular development experience. It provides a curated
+          collection of Angular components, directives, services, and utilities,
+          all carefully crafted to accelerate your development workflow and
+          promote best practices.
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/getting-started/components/introduction/introduction.component.html:13]: ok

File: projects/arc-docs/src/styles.scss [hunk 7]
```diff
@@ -1 +1,316 @@
...
+  font-family: #{nb-theme(font-family-primary)};
+  font-weight: 600;
+  line-height: 1.5rem;
+  background-color: map.get($font, 'light');
+  border: 0.063rem solid map.get($color, 'border-disable');
+  position: absolute;
+
+  &:hover {
+    border-color: map-get($font, 'placeholder');
+  }
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-docs/src/styles.scss:78]: removed

File: projects/arc-lib/src/lib/components/header /header.component.html
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/header /header.component.html:1]: done

File: projects/arc/src/app/main/main.component.ts (hunk 3 above)
Comment (Deepika516) [reply projects/arc/src/app/main/main.component.ts:66]: bcos i created the standalone component of this, i removed this from here , added this in header comp

File: projects/arc-docs/src/app/docs/docs.component.scss [hunk 8]
```diff
@@ -1,3 +1,71 @@
+@use 'sass:map';
+@use '../../../../arc-lib/src/lib/theme/styles/variables' as *;
```
Comment (devyaniofficial) [inline projects/arc-docs/src/app/docs/docs.component.scss:2]: add this in angular json 
"stylePreprocessorOptions": {
              "includePaths": [
                "src/app/theme"
              ]
            }, and use it like @import 'element-override'; in scss files

File: projects/arc-docs/src/app/docs/docs.component.scss [hunk 9]
```diff
@@ -1,3 +1,71 @@
...
+@use '@nebular/theme/styles/theming' as *;
+@use '../../styles.scss' as *;
```
Comment (devyaniofficial) [inline projects/arc-docs/src/app/docs/docs.component.scss:4]: same

File: projects/arc-docs/src/app/docs/docs.component.scss [hunk 10]
```diff
@@ -1,3 +1,71 @@
...
+    .logo {
+      padding: 0 1.25rem;
+      padding-left: 0.625rem;
+      font-size: 1.75rem;
+      white-space: nowrap;
+      text-decoration: none;
+    }
+  }
+}
+.hi {
```
Comment (devyaniofficial) [inline projects/arc-docs/src/app/docs/docs.component.scss:66]: this class name seems incorrect

File: projects/arc-docs/src/app/docs/docs.component.ts [hunk 11]
```diff
@@ -1,8 +1,67 @@
...
+    private readonly authService: AuthService,
+    private readonly menuService: NbMenuService,
+    private readonly iconMgr: IconPacksManagerService,
+    private router: Router,
+  ) {
+    super(route, location);
+    this.iconMgr.registerSvgs();
+  }
+
+  menu: NbMenuItem[] = [
```
Comment (devyaniofficial) [inline projects/arc-docs/src/app/docs/docs.component.ts:32]: create a separate file for this constant and remove commented code

File: projects/arc-docs/src/app/docs/docs.component.ts [hunk 12]
```diff
@@ -1,8 +1,67 @@
...
+      ],
+    },
+  ];
+
+  ngOnInit(): void {
+    this.menuService
+      .onItemClick()
+      .pipe(takeUntil(this._destroy$))
+      .subscribe(menu => {
+        console.log(menu);
```
Comment (devyaniofficial) [inline projects/arc-docs/src/app/docs/docs.component.ts:64]: nothing is happening in this function, remove

File: projects/arc-docs/src/app/docs/docs.module.ts [hunk 13]
```diff
@@ -3,9 +3,33 @@ import {CommonModule} from '@angular/common';
...
+    HeaderComponent,
+    SidebarComponent,
+    CoreAuthModule,
+  ],
+  providers: [
+    NgxPermissionsStore,
+    {
+      provide: APP_CONFIG,
+      useValue: environment,
+    },
```
Comment (devyaniofficial) [inline projects/arc-docs/src/app/docs/docs.module.ts:32]: this should not be needed from all the modules

File: projects/arc-docs/src/styles.scss [hunk 14]
```diff
@@ -1 +1,281 @@
 /* You can add global styles to this file, and also import other style files */
+@use '../../arc/src/themes.scss' as *;
+@use 'sass:map';
+@use '../../arc-lib/src/lib/theme/styles/index' as *;
+@use '@nebular/theme/styles/globals' as *;
+@use '@nebular/auth/styles/globals' as *;
+@use '../../arc-lib/src/lib/theme/styles/variables' as *;
```
Comment (devyaniofficial) [inline projects/arc-docs/src/styles.scss:7]: fix all of these

File: projects/arc-lib/src/lib/components/header/header.component.scss [hunk 15]
```diff
@@ -0,0 +1,31 @@
+@use 'sass:map';
+@use '../../theme/styles/variables' as *;
+@use '../.././../../../arc/src/styles.scss' as *;
+@use '../../theme/styles/mixins' as *;
```
Comment (devyaniofficial) [inline projects/arc-lib/src/lib/components/header/header.component.scss:4]: same

File: projects/arc-lib/src/lib/components/sidebar/sidebar.component.ts [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+    private readonly sidebarService: NbSidebarService,
+    private readonly authService: AuthService,
+    private readonly menuService: NbMenuService,
+    private readonly iconMgr: IconPacksManagerService,
+    private router: Router,
+  ) {
+    super(route, location);
+    this.iconMgr.registerSvgs();
+  }
+  menu: NbMenuItem[] = [
```
Comment (devyaniofficial) [inline projects/arc-lib/src/lib/components/sidebar/sidebar.component.ts:43]: move it to separate file

**Review by devyaniofficial — CHANGES_REQUESTED** [review]

File: projects/arc-docs/src/app/docs/docs.component.scss (hunk 10 above)
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/docs.component.scss:66]: removed

File: projects/arc-docs/src/app/docs/docs.component.ts (hunk 11 above)
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/docs.component.ts:32]: ok

File: projects/arc-docs/src/app/docs/docs.component.ts (hunk 12 above)
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/docs.component.ts:64]: ok removed

File: projects/arc-docs/src/app/docs/docs.module.ts (hunk 13 above)
Comment (Deepika516) [reply projects/arc-docs/src/app/docs/docs.module.ts:32]: ok

File: projects/arc-lib/src/lib/components/sidebar/sidebar.component.ts (hunk 16 above)
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/sidebar/sidebar.component.ts:43]: done

**Review by devyaniofficial — APPROVED** [review]

File: projects/arc-docs/src/app/docs/docs.module.ts (hunk 13 above)
Comment (yeshamavani) [reply projects/arc-docs/src/app/docs/docs.module.ts:29]: ???

File: projects/arc-docs/src/app/docs/getting-started/components/introduction/introduction.component.html [hunk 17]
```diff
@@ -1 +1,153 @@
...
+      <nb-card-header fixed>
+        <h2 class="nb-card-header-title">Key Features:</h2>
+      </nb-card-header>
+      <nb-card-body>
+        <div class="list-wrapper">
+          <ul>
+            <li class="list-wrap">
+              <strong>Angular Components:</strong> Explore a diverse range of
+              pre-built Angular components, including SignIn,
+              SignUp,forgetPassword,ResetPassword, modals, and more. These
```
Comment (yeshamavani) [inline projects/arc-docs/src/app/docs/getting-started/components/introduction/introduction.component.html:43]: spelling of models

**Review by yeshamavani — APPROVED** [review]

**Pull request merged**